### PR TITLE
Feat/performance page endpoints

### PR DESCRIPTION
**Pull request by tomasgaudino**
### Enhancements and Updates
This PR introduces the following enhancements and updates:

#### Database parsing and management:

- List and read databases
- Create, list and read checkpoints

#### Performance processing:

- Get performance results through backtesting engine

### CODE REVIEW

**Comment by tomasgaudino** [comment]
@cardosofede ready for review

**Comment by tomasgaudino** [comment]
@nikspz @rapcmia hey guys can you take a look to this pr please?

**Comment by rapcmia** [comment]
> @nikspz @rapcmia hey guys can you take a look to this pr please?

Hi @tomasgaudino im currently blocked by an issue i found on my local machine and created a ticket so others to confirm https://github.com/hummingbot/backend-api/issues/44, other than that i am able to setup and run backtesting and manually create container using `:8000/docs`, will update more of my tests later tonight

**Comment by rapcmia** [comment]
PR update:
- Setup this PR with development dashboard and broker
- Compare this PR with development branch
  - Found new endpoints for market performance and database management
- Deployed an instance and wait for few trades to happened
- Run endpoints after getting few trades
- Database:
   - post /list-databases 
     image
   - post /read-databases 
    image
   - post /create-checkpoints 
     - create a checkpoint sqlite file and saved on `./bots/data` folder
   - post /list-checkpoints
     - Displays all available checkpoint on ./bots/data folder 
     - Toggle between false and true to display full dir ok
   - post /load-checkpoint 
     - Load the dir path of the checkpoint
     - Displays executors, orders, trade_fill and controller tables 
      image

--- 

For post /get-performance-results
image
 - Can you advise a sample response schema to use?

File: bots/conf/controllers/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*
```
Comment (cardosofede) [inline bots/conf/controllers/.gitignore:1]: this is not necessary

**Review by cardosofede — CHANGES_REQUESTED** [review]
remove the * and it's good to go @rapcmia let's continue with the review after @tomasgaudino answers

**Comment by rapcmia** [comment]
PR update:
- Ran tests with https://github.com/hummingbot/dashboard/pull/184
- When trying to load `strategy performance` backend-api crashed
  ```
  INFO:     127.0.0.1:54484 - "GET /get-active-bots-status HTTP/1.1" 200 OK
  INFO:     127.0.0.1:54488 - "POST /list-databases HTTP/1.1" 500 Internal Server Error
  ERROR:    Exception in ASGI application
  Traceback (most recent call last):
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi
      result = await app(  # type: ignore[func-returns-value]
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 60, in __call__
      return await self.app(scope, receive, send)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/fastapi/applications.py", line 1054, in __call__
      await super().__call__(scope, receive, send)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/applications.py", line 113, in __call__
      await self.middleware_stack(scope, receive, send)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/middleware/errors.py", line 187, in __call__
      raise exc
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/middleware/errors.py", line 165, in __call__
      await self.app(scope, receive, _send)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
      await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
      raise exc
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
      await app(scope, receive, sender)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/routing.py", line 715, in __call__
      await self.middleware_stack(scope, receive, send)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/routing.py", line 735, in app
      await route.handle(scope, receive, send)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/routing.py", line 288, in handle
      await self.app(scope, receive, send)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/routing.py", line 76, in app
      await wrap_app_handling_exceptions(app, request)(scope, receive, send)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
      raise exc
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
      await app(scope, receive, sender)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/starlette/routing.py", line 73, in app
      response = await f(request)
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/fastapi/routing.py", line 301, in app
      raw_response = await run_endpoint_function(
    File "/Users/rapcomia/miniconda3/envs/backend-api/lib/python3.10/site-packages/fastapi/routing.py", line 212, in run_endpoint_function
      return await dependant.call(**values)
    File "/Users/rapcomia/github/hummingbot/backend-api/36/routers/manage_databases.py", line 19, in list_databases
      return file_system.list_databases()
    File "/Users/rapcomia/github/hummingbot/backend-api/36/utils/file_system.py", line 207, in list_databases
      archived_instances = self.list_folders("archived")
    File "/Users/rapcomia/github/hummingbot/backend-api/36/utils/file_system.py", line 47, in list_folders
      return [d for d in os.listdir(dir_path) if os.path.isdir(os.path.join(dir_path, d))]
  FileNotFoundError: [Errno 2] No such file or directory: 'bots/archived'
  ```

Steps to reproduce:
1. Setup this PR with https://github.com/hummingbot/dashboard/pull/184 and broker
2. Go to Strategy Performance page

**Review by cardosofede — APPROVED** [review]

**Pull request merged**